packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -0,0 +1,1582 @@
+                minAmountLD: amount,
+                extraOptions: extraOptions,
+                composeMsg: composeMsg,
+                oftCmd: oftCmd
+            });
+    }
+
+    /**
+     * @dev Build SendParam struct
+     */
+    function _buildSendParam(
+        uint16 destinationChainId,
+        address destinationAdapter,
+        uint256 amount,
+        bytes memory composeMsg,
+        BridgeTypes.AdapterParams calldata
+    ) internal view returns (SendParam memory) {
+        // Always use taxi mode for compose functionality and reliability
+        bytes memory oftCmd = OftCmdHelper.taxi(); // Always use taxi mode like in the example
+
+        // Add compose options when compose message is present
+        bytes memory extraOptions = composeMsg.length > 0
+            ? OptionsBuilder.newOptions().addExecutorLzComposeOption(
+                0,
+                uint128(composeGasLimit),
+                0
+            )
+            : bytes("");
+
+        return
+            SendParam({
+                dstEid: chainToEndpointId[destinationChainId],
+                to: destinationAdapter.toBytes32(),
+                amountLD: amount,
+                minAmountLD: amount,
+                extraOptions: extraOptions,
+                composeMsg: composeMsg,
+                oftCmd: oftCmd // Always "" for taxi mode
+            });
+    }
+
+    /**
+     * @dev Update minimum amount based on quote with proper validation
+     */
+    function _updateMinAmount(
+        IStargateV2 stargate,
+        SendParam memory sendParam,
+        uint256 amount
+    ) internal view {
+        try stargate.quoteOFT(sendParam) returns (
+            OFTLimit memory oftLimit,
+            OFTFeeDetail[] memory,
+            OFTReceipt memory oftReceipt
+        ) {
+            // Validate OFT limits first
+            if (amount < oftLimit.minAmountLD) {
+                revert InsufficientAmount(amount, oftLimit.minAmountLD);
+            }
+            if (amount > oftLimit.maxAmountLD) {
+                revert ExceedsMaxAmount(amount, oftLimit.maxAmountLD);
+            }
+
+            // Validate received amount
+            if (oftReceipt.amountReceivedLD == 0) {
+                revert ZeroAmountReceived();
+            }
+
+            // Check that received amount is not higher than input (suspicious)
+            if (oftReceipt.amountReceivedLD > amount) {
+                revert InvalidAmountReceived(
+                    oftReceipt.amountReceivedLD,
+                    amount
+                );
+            }
+
+            // Calculate minimum slippage threshold (use configurable tolerance)
+            uint256 minExpectedAmount = (amount *
+                (10000 - slippageToleranceBps)) / 10000;
+
+            // Ensure received amount is within acceptable slippage
+            if (oftReceipt.amountReceivedLD < minExpectedAmount) {
+                // Use fallback calculation instead of the quote
```

Comment:
I don't thing the execution should be continued if the bridge says the slippage is too high - best would be to revert